azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunner.java

```diff
@@ -917,15 +920,22 @@ private Boolean isConditionOnRuntimeVariableMet(final ExecutableNode node) {
       return true;
     }
 
-    final Matcher matcher = CONDITION_VARIABLE_REPLACEMENT_PATTERN.matcher(condition);
     String replaced = condition;
+    // Replace the condition on job status macro with "true" to skip the evaluation by Script
+    // Engine since it has already been evaluated.
+    Matcher matcher = CONDITION_ON_JOB_STATUS_PATTERN.matcher(condition);
+    if (matcher.find()) {
+      replaced = condition.replace(matcher.group(1), "true");
+    }
+
+    matcher = CONDITION_VARIABLE_REPLACEMENT_PATTERN.matcher(replaced);
```

Comment:
Minor thing but maybe it'd be clearer to use a different name for the different matchers here, and thus allowing both variables to be final.

Reply:
Sure, I'll update the matcher name, thanks!